docs/source/sat_solving.rst

```diff
@@ -45,3 +45,58 @@ to the right of it. ``*`` is a wild-card that matches any version.
 ``== 1.1.1-1``    ``>= 1.1.1-1``   ``<= 1.1.1-1``   ``*``
 ``!= 1.1.1-1``    ``!= 1.1.1-1``   ``!= 1.1.1-1``   ``!= 1.1.1-1``
 ===============  ===============   ===============  ===============
+
+
+Requirements
+------------
+
+There are currently three different ``Requirement`` classes:
+:class:`Requirement<simplesat.constraints.requirement.Requirement>`,
+:class:`InstallRequirement
+<simplesat.constraints.requirement.InstallRequirement>` and
+:class:`ConflictRequirement
+<simplesat.constraints.requirement.ConflictRequirement>`. They have no internal
+differences, but this split allows us to reliably track the origin of a
+requirement via its type and avoid using it in an inappropriate context.
+
+We care about the difference between a requirement created from
```

Comment:
Who is responsible for creating requirements? Is this done by external code using the solver? If it's done by a particular class or part of the solver process, can you mention that here?